src/common.ts

```diff
@@ -59,23 +77,64 @@ const createHashMapping = (disclosures: Disclosure[], hash_alg: string): SdDiges
   return map;
 };
 
+/**
+ * Iterates through an array
+ * inserts claim if disclosed
+ * removes any undisclosed claims
+ */
+const unpackArray = ({ arr, map }) => {
+  const unpackedArray = [];
+  arr.forEach((item) => {
+    if (item instanceof Object) {
+      // if Array item is { '...': <SD_HASH_DIGEST> }
+      if (item[SD_LIST_PREFIX]) {
+        const hash = item[SD_LIST_PREFIX];
+        const disclosed = map[hash];
+        // Only add item if disclosed
```

Comment:
@ragnika code is quite self explanatory in a lot of cases and naming is good. No need to leave a comment for every line.

Reply:
@linasi Cheers, some were pseudo code to help me write this recursive function.
I'll clean up the redundant comments up in the next PR